### PR TITLE
Pull request for #82: Update docs for 4.1.0 

### PR DESCRIPTION
**Pull request by arosenbe**
I've updated the Template readme and gitignore. I've also added an example of our anything builder. Can you let me know if you see any misinformation, typos, features I forgot, etc. 

This review should also cover the changes I made to the ra manual wiki at branch `template#82_docs`.

@yuchuan2016 and @Shun-Yang should feel free to review as well. 

For #82. See https://github.com/gslab-econ/template/issues/82#issuecomment-326415994 for checklist.

### CODE REVIEW

**Comment by stanfordquan** [comment]
On it.

File: README.md [hunk 1]
```diff
@@ -53,73 +54,126 @@ If you want to create a repository with the same structure as this template you
...
 
 * When a large dataset is stored externally, there are a few options. 
     * The top-level readme can specify manual download and storage instructions. This is simple, easy to customize, and unlikely to cause errors during a SCons build. It does, however, require each user to successfully download the same dataset, perhaps in an unstructured manner. 
-    * The download can be incorporated into the SCons build. We either execute a program to transfer data (e.g., `rsync` or `rclone`) directly in a standard SCons command or from within a script executed by one of our custom builders. These methods have the benefits of automation and dependency tracking, but they can introduce idiosyncratic errors if the download steps are prone to failure.
+    * The download can be incorporated into the SCons build. We either execute a program to transfer data (e.g., `rsync` or `rclone`) using our "anything build" or from within a script executed by one of our other custom builders. These methods have the benefits of automation and dependency tracking, but they can introduce idiosyncratic errors if the download steps are prone to failure.
```
Comment (stanfordquan) [inline README.md]: `_build_anything` function instead of "anything build"?

File: README.md [hunk 2]
```diff
@@ -53,73 +54,126 @@ If you want to create a repository with the same structure as this template you
...
-- If you're using Matlab, make sure it's been added to your PATH, and that you have installed the Matlab YAML parser following the instruction [here](https://github.com/gslab-econ/gslab_matlab/blob/f11eff492e0c982cf344c60b7e7ce0e7b7a66872/README.md#installation-instructions-for-matlab-r2016b).
+We recommend that you manually copy the desired files or directories from the `release` directory in `analysis` to a directory called `input` in `paper_slides`. We uncouple these top-level subdirectories to compartmentalize the research process. A busy coauthor can build `paper_slides` in a reproducible manner without wrangling data or repeating a lengthy analysis.  
 
-##### Can I pass "command line style" arguments to a script?
+There are lighter-weight methods to connect these subdirectories, but they may make your project more prone to errors.
+* Create a symlink between `analysis/release` and `paper_slides/input` (not platform-independent).
+* Point to `analysis/release` from the `config_global.yaml` in `paper_slides` (hard to parse YAML in LyX and LaTeX).
+
```
Comment (stanfordquan) [inline README.md]: Does

```python
 env.Install('../paper-slides/input', '#release/') 
```

work?

File: README.md [hunk 3]
```diff
@@ -53,73 +54,126 @@ If you want to create a repository with the same structure as this template you
...
+#### What software can I use for data analysis?
+
+We have custom builders for R, Stata, and MATLAB. They all use the same syntax as the Python builder. You'll need to add the builders to the SCons environment by uncommenting their definitions in the SConstruct. Also check that the executable has been added to your PATH.
```
Comment (stanfordquan) [inline README.md]: Python, R, Stata, and MATLAB.

File: README.md [hunk 4]
```diff
@@ -7,31 +7,33 @@ The GSLab Template is a minimal working demonstration of the tools and organizat
...
 * [LyX](https://www.lyx.org/Download) (with instructions for LaTeX) 
     * Add LyX to your PATH for [Windows](http://www.computerhope.com/issues/ch000549.htm), [Mac](http://hathaway.cc/post/69201163472/how-to-edit-your-path-environment-variables-on-mac), and [Linux](http://stackoverflow.com/questions/14637979/how-to-permanently-set-path-on-linux).
     * The beamer theme [`metropolis`](https://github.com/matze/mtheme). This is part of MikTeX since Dec 2014. 
 
-## Getting started
+## Quick start
 
 1. Open a shell, clone the repository, and navigate to its root.
     ```bash
     git clone https://github.com/gslab-econ/template.git
```
Comment (stanfordquan) [inline README.md]: I don't know where we changed cloning protocol from SSH to HTTPS, but we should be consistent (with the `Copying the template` bit.

Comment (stanfordquan) [reply README.md]: I've also had issue cloning the repo. Let's have a bit in FAQ saying "If you have this error then try `git lfs clone`."

```bash
Errors logged to /Users/lequan/ssh_template/.git/lfs/objects/logs/20170904T151244.091215032.log
Use `git lfs logs last` to view the log.
error: external filter 'git-lfs smudge -- %f' failed 2
error: external filter 'git-lfs smudge -- %f' failed
fatal: release/paper/text.pdf: smudge filter lfs failed
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry the checkout with 'git checkout -f HEAD'

```

File: analysis/source/prepare_data/SConscript [hunk 5]
```diff
@@ -19,4 +20,11 @@ env.BuildPython(target  = ['#%s/data.txt'       % build],
 #                 source  = ['#%s/create_data.m'   % source],
 #                 log_ext =  'create_data')
 
+# # Not Windows
```
[the remark this answers is not in the document]
Comment (stanfordquan) [reply analysis/source/prepare_data/SConscript]: What I mean is we should have some documentation of how to use `_build_anything` the way we document `build_R`

File: README.md (hunk 1 above)
Comment (arosenbe) [reply README.md]: Meant to say "anything builder" to mirror e.g., Python builder. You cool with that?

File: README.md (hunk 2 above)
Comment (arosenbe) [reply README.md]: Should work, will note.

File: analysis/source/prepare_data/SConscript [hunk 6]
```diff
@@ -20,10 +20,9 @@ env.BuildPython(target  = ['#%s/data.txt'       % build],
 #                 source  = ['#%s/create_data.m'   % source],
 #                 log_ext =  'create_data')
 
-# # Not Windows
```
Comment (arosenbe) [inline analysis/source/prepare_data/SConscript]: Why'd we cut this? The shell script won't run on Windows.

File: analysis/source/prepare_data/create_data.sh [hunk 7]
```diff
@@ -0,0 +1 @@
+echo x > build/prepare_data/data.txt; seq 1 300000 >> build/prepare_data/data.txt
```
Comment (arosenbe) [inline analysis/source/prepare_data/create_data.sh]: Split over two lines.

File: README.md [hunk 8]
```diff
@@ -7,31 +7,33 @@ The GSLab Template is a minimal working demonstration of the tools and organizat
...
     cd template
```
[the remark this answers is not in the document]
Comment (arosenbe) [reply README.md]: Why?

[the remark this answers is not in the document]
Comment (stanfordquan) [reply README.md]: I can't imagine someone keeping a folder named "template" to work on as a project. Might as well provide the syntax.

File: README.md (hunk 1 above)
Comment (stanfordquan) [reply README.md]: let's capitalized/distinguish "anything", or anything-builder.

File: analysis/source/prepare_data/SConscript (hunk 6 above)
Comment (stanfordquan) [reply analysis/source/prepare_data/SConscript]: Ah I see, the builder works in Windows but not the script. Feel free to add back, my bad.

File: analysis/source/prepare_data/create_data.sh (hunk 7 above)
Comment (stanfordquan) [reply analysis/source/prepare_data/create_data.sh]: Fine with me

File: README.md (hunk 1 above)
Comment (arosenbe) [reply README.md]: Capitalized A in anything.

File: README.md (hunk 4 above)
Comment (arosenbe) [reply README.md]: Updated default to  
```
git lfs clone
```

**Comment by arosenbe** [comment]
@stanfordquan, I think I responded to all your requests. Can you look again?

**Comment by stanfordquan** [comment]
Looks good, thanks! Feel free to merge. I have a local merge commit on the RA manual and will push it myself.

**Comment by stanfordquan** [comment]
I made the edits in the RA manual as of commit `0271a40`.
